File: SelfSchoolBE/SelfSchoolDAL/Entities/Teacher.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SelfSchoolDAL.Entities
{
    public class Teacher
    {
        [Key]
        public int idTeacher { get; set; }
        public string loginTeacher { get; set; }
        public string passwordTeacher { get; set; }
        public string nameTeacher { get; set; }
        public string surnameTeacher { get; set; }
        public long birthdayTeacher { get; set; }
        public string emailTeacher { get; set; }
        public string phoneTeacher { get; set; }
        public string subjectTeacher { get; set; }
    }
}

File: SelfSchoolFE/src/app/classes/teacher.ts
import {ClassSchool} from "./class-school";

export class Teacher {
  private readonly _idTeacher: number = 0;
  private readonly _loginTeacher: string = '';
  private readonly _passwordTeacher: string = '';
  private readonly _nameTeacher: string = '';
  private readonly _surnameTeacher: string = '';
  private readonly _birthdayTeacher: number = 0;
  private readonly _emailTeacher: string = '';
  private readonly _phoneTeacher: string = '';
  private readonly _subjectTeacher: string = '';

  constructor(idTeacher: number, loginTeacher: string, passwordTeacher: string, nameTeacher: string,
              surnameTeacher: string, birthdayTeacher: number, emailTeacher: string, phoneTeacher: string,
              subjectTeacher: string) {
    this._idTeacher = idTeacher;
    this._loginTeacher = loginTeacher;
    this._passwordTeacher = passwordTeacher;
    this._nameTeacher = nameTeacher;
    this._surnameTeacher = surnameTeacher;
    this._birthdayTeacher = birthdayTeacher;
    this._emailTeacher = emailTeacher;
    this._phoneTeacher = phoneTeacher;
    this._subjectTeacher = subjectTeacher;
  }

  public get idTeacher(): number {
    return this._idTeacher;
  }

  public get loginTeacher(): string {
    return this._loginTeacher;
  }

  public get passwordTeacher(): string {
    return this._passwordTeacher;
  }

  public get nameTeacher(): string {
    return this._nameTeacher;
  }

  public get surnameTeacher(): string {
    return this._surnameTeacher;
  }

  public get birthdayTeacher(): number {
    return this._birthdayTeacher;
  }

  public get emailTeacher(): string {
    return this._emailTeacher;
  }

  public get phoneTeacher(): string {
    return this._phoneTeacher;
  }

  public get subjectTeacher(): string {
    return this._subjectTeacher;
  }
}
